src/openrct2/ride/Vehicle.cpp

```diff
@@ -2363,61 +2363,20 @@ static void vehicle_update_waiting_for_passengers(rct_vehicle* vehicle)
                 return;
             }
 
+            // any load: load=4 , full: load=3,  3/4s: load=2, half: load=2, quarter: load=1
             uint8_t load = ride->depart_flags & RIDE_DEPART_WAIT_FOR_LOAD_MASK;
-            if (load == 3)
-            {
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
-
-            uint8_t three_quater_seats = (3 * num_seats_on_train) / 4;
-            if (three_quater_seats != 0 && num_peeps_on_train >= three_quater_seats)
-            {
-                vehicle->update_flags |= VEHICLE_UPDATE_FLAG_TRAIN_READY_DEPART;
-                train_ready_to_depart(vehicle, num_peeps_on_train, num_used_seats_on_train);
-                return;
-            }
+            double peepTarget = (((double)load + 1) * num_seats_on_train) / 4;
```

Comment:
This can actually yield different results on different cpus/platforms, use integers only.